lib/Sema/TypeChecker.cpp

```diff
@@ -231,29 +231,6 @@ ModuleDecl *TypeChecker::getStdlibModule(const DeclContext *dc) {
   return StdlibModule;
 }
 
-Type TypeChecker::lookupBoolType(const DeclContext *dc) {
-  if (!boolType) {
-    boolType = ([&] {
-      SmallVector<ValueDecl *, 2> results;
-      getStdlibModule(dc)->lookupValue({}, Context.getIdentifier("Bool"),
-                                       NLKind::QualifiedLookup, results);
-      if (results.size() != 1) {
-        diagnose(SourceLoc(), diag::broken_bool);
-        return Type();
-      }
-
-      auto tyDecl = dyn_cast<NominalTypeDecl>(results.front());
-      if (!tyDecl) {
-        diagnose(SourceLoc(), diag::broken_bool);
```

Comment:
If we're not going to bother diagnosing this anymore, you can remove the diagnostic too.

Reply:
We use this diagnostic elsewhere in [TypeCheckDeclObjC](https://github.com/apple/swift/blob/a1d38c1012a47c548310181f03e212b3fbfee806/lib/Sema/TypeCheckDeclObjC.cpp#L586) and with `Bool.init(_builtinBooleanLiteral:)` in [CSApply](https://github.com/apple/swift/blob/2ea00c38156826cb8e9d8d5ef9d2a55ba5592380/lib/Sema/CSApply.cpp#L3220)

Reply:
Although arguably the last one could be its own diagnostic. These broken diagnostics are something I want to improve in the future however.

Reply:
I think the 'broken stdlib' diagnostics should become fatal. Parts of the compiler don't recover well from these and assert later, so we may as well just exit early when it happens.

Reply:
Are you suggesting something like `llvm_unreachable` for these 'broken stdlib' diagnostics?

Reply:
I definitely have run into cases where the failures are pretty terrible and at times you just get weird things like the literal doesn't conform to equatable or some equally odd until you look through and find special look ups in the compiler.

Reply:
I think for this patch I'll go ahead and add back the diagnostic, but in a follow up I'll fix up these broken diagnostics to be fatal.